src/uct/tcp/tcp_cm.c

```diff
@@ -511,31 +518,60 @@ unsigned uct_tcp_cm_handle_conn_pkt(uct_tcp_ep_t **ep, void *pkt, uint32_t lengt
     return 0;
 }
 
-unsigned uct_tcp_cm_conn_progress(uct_tcp_ep_t *ep)
+ucs_status_t uct_tcp_cm_conn_complete(uct_tcp_ep_t *ep,
+                                      unsigned *progress_count)
 {
+    unsigned count = 0;
     ucs_status_t status;
 
-    if (!ucs_socket_is_connected(ep->fd)) {
-        ucs_error("tcp_ep %p: connection establishment for "
-                  "socket fd %d was unsuccessful", ep, ep->fd);
-        goto err;
+    status = uct_tcp_cm_send_magic_number(ep);
+    if (status != UCS_OK) {
+        goto out;
     }
 
+    count++;
+
     status = uct_tcp_cm_send_event(ep, UCT_TCP_CM_CONN_REQ);
     if (status != UCS_OK) {
-        return 0;
+        goto out;
     }
 
+    count++;
+
     uct_tcp_cm_change_conn_state(ep, UCT_TCP_EP_CONN_STATE_WAITING_ACK);
     uct_tcp_ep_mod_events(ep, UCS_EVENT_SET_EVREAD, 0);
 
     ucs_assertv((ep->tx.length == 0) && (ep->tx.offset == 0) &&
                 (ep->tx.buf == NULL), "ep=%p", ep);
-    return 1;
+
+out:
+    if (progress_count != NULL) {
+        *progress_count += count;
+    }
+    return UCS_OK;
+}
+
+unsigned uct_tcp_cm_conn_progress(uct_tcp_ep_t *ep)
+{
+    unsigned progress_count = 0;
+    ucs_status_t status;
+
+    if (!ucs_socket_is_connected(ep->fd)) {
+        ucs_error("tcp_ep %p: connection establishment for "
+                  "socket fd %d was unsuccessful", ep, ep->fd);
+        goto err;
+    }
+
+    status = uct_tcp_cm_conn_complete(ep, &progress_count);
+    if (status != UCS_OK) {
+        return status;
```

Comment:
function is supposed to return `unsigned` not `ucs_status_t`

Reply:
fixed